api/post.go

```diff
@@ -670,8 +670,15 @@ func sendNotifications(c *Context, post *model.Post, team *model.Team, channel *
 							alreadySeen[session.DeviceId] = session.DeviceId
 
 							msg := model.PushNotification{}
-							msg.Badge = 1
+							if badge := <-Srv.Store.User().GetUnreadCount(id); badge.Err == nil {
```

Comment:
Traditionally we do this the other way around. We check if Err != nil and do the error case first. I don't know why but I am used to it now. Can you flip this please?